bug: nil pointer on verify proof offchain CSP

```
2022-10-10T11:11:09Z	DEBUG	vochain/transaction.go:560	new vote 379eb910c0ff4abda948c2fa43390c37a472e3b78ec4da866ae9821c8cb8630c for address 0x9F4D681E3cBC6f858A24e14EdA09F17a54bC7ad6 and process 24ba45c1dfaa4a081070e9d555b5d19629a6bcc8b77f4ae6d39c03000000001c
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x2b57358]

goroutine 234740198 [running]:
go.vocdoni.io/dvote/vochain.VerifyProofOffChainCSP(0xcd?, 0xc00c5a1780, 0x3?, {0xc0177a25c0, 0x40, 0x40}, {0xc019283720, 0x20, 0x20}, {0xc00585a3f0, ...}, ...)
	go.vocdoni.io/dvote/vochain/proof.go:114 +0xb8
go.vocdoni.io/dvote/vochain.VerifyProof(0xc00b308591?, 0xc0041972b8?, 0x118ae54?, {0xc004197320?, 0x16f01d3?, 0xc00b308591?}, {0x1?, 0xf?, 0x0?}, {0xc00585a3f0, ...}, ...)
	go.vocdoni.io/dvote/vochain/proof.go:55 +0xd2
go.vocdoni.io/dvote/vochain.(*BaseApplication).VoteEnvelopeCheck(0xc0051f8210, 0xc0148e4c60, {0xc017bbba80, 0x73, 0x73}, {0xc01b337590, 0x41, 0x50}, {0xa3, 0x51, ...}, ...)
	go.vocdoni.io/dvote/vochain/transaction.go:562 +0x10f4
```

Reported on gw1 dev

--- a/vochain/proof.go
+++ b/vochain/proof.go
@@ -52,6 +52,9 @@ func VerifyProof(process *models.Process, proof *models.Proof,
 	default:
 		return false, nil, fmt.Errorf("census origin not compatible")
 	}
+	if process == nil || proof == nil {
+		return false, nil, fmt.Errorf("nil process or proof: %v / %v", process, proof)
+	}
 	valid, weight, err := verifyProof(process, proof,
 		process.CensusOrigin, process.CensusRoot, process.ProcessId,
 		pubKey, addr)
@@ -111,6 +114,9 @@ func VerifyProofOffChainCSP(process *models.Process, proof *models.Proof,
 	key := addr.Bytes()
 
 	p := proof.GetCa()
+	if p == nil {
+		return false, nil, fmt.Errorf("invalid CA proof")
+	}
 	if !bytes.Equal(p.Bundle.Address, key) {
 		return false, nil, fmt.Errorf(
 			"CSP bundle address and key do not match: %x != %x", key, p.Bundle.Address)

--- a/vochain/transaction.go
+++ b/vochain/transaction.go
@@ -528,6 +528,9 @@ func (app *BaseApplication) VoteEnvelopeCheck(ve *models.VoteEnvelope, txBytes, 
 		if ve.Proof == nil {
 			return nil, voterID.Nil(), fmt.Errorf("proof not found on transaction")
 		}
+		if ve.Proof.Payload == nil {
+			return nil, voterID.Nil(), fmt.Errorf("invalid proof payload provided")
+		}
 
 		// If process encrypted, check the vote is encrypted (includes at least one key index)
 		if process.EnvelopeType.EncryptedVotes {

